### DISCUSSION TITLE
c++ sdk vs others(e.g. go sdk) for gcloud

### DISCUSSION THREAD

**Discussion by laoshaw** [googleapis/google-cloud-cpp#9518] [General]
If I have to pick one that is complete(covers most services) and solid, which one should I pick? what's their difference?

AWS has a near complete c++ SDK for all its services but a not-so-complete go-SDK(only supports a few services), gcloud seems the opposite, that go-sdk supports near-complete services but not c++-sdk, am I missing something?
Thanks,

**Comment by coryan** [comment]
I am not sure I can make a general recommendation on selecting a programming language and SDK.  I expect that is a complex decision that involves what other libraries you want to use, the experience of your team, any pre-existing code you may have, etc. etc.

As to whether the C++ SDK is "complete".  I think we cover most services at this time, the big exceptions are:

- Google Compute Engine (we are trying to get this later this year).  Note that this is the *admin* API for GCE, you can deploy C++ applications to GCE.
- Some BigQuery, notably the APIs to query tables, create tables, etc.  We expect most admin operations around the end of the year too.
- Firestore.  We have no concrete dates for this, but we do want to support it.

If there are other services that are missing, let us know, we can generate libraries for most services relatively quickly.

**Reply by laoshaw** [reply]
very helpful, thanks! wish google has a feature/service-supported comparison table for its SDKs
